app/js/components/manageApps/Login.jsx

```diff
@@ -38,7 +39,15 @@ class Login extends Component {
         'Authorization': auth
       }
     }).then(response => {
-      (response.data.authenticated) ? hashHistory.push('/') : toastr.error('Invalid username or password');
+        if (response.data.authenticated) {
+          if (redirectUrl) {
+            window.location = redirectUrl;
+          } else {
+            hashHistory.push("/");
```

Comment:
i think you could achieve the same thing with two lines of code as ==>     
`response.data.authenticated ? (redirectUrl ? window.location = redirectUrl : hashHistory.push("/")) : toastr.error("Invalid username or password");`

Reply:
@justmesam I initially used the ternary operators but they made the code less readable. Decided against it in this case unless It becomes the codebase's convention